src/controllers/post/searchPosts.ts

```diff
@@ -0,0 +1,177 @@
+  body: z.object({
+    fromPlace: z
+      .nativeEnum(Place, {
+        invalid_type_error: "fromPlace must be a valid enum of the defined places"
+      })
+      .optional(),
+
+    toPlace: z
+      .nativeEnum(Place, {
+        invalid_type_error: "toPlace must be a valid enum of the defined places"
+      })
+      .optional(),
+
+    startTime: z
+      .coerce.date({
+        invalid_type_error: "startTime must be a Date() object"
+      })
+      .optional(),
+
+    endTime: z
+      .coerce.date({
+        invalid_type_error: "endTime must be a Date() object"
+      })
+      .optional(),
+
+    availableSeats: z
+      .number({
+        invalid_type_error: "availableSeats must be an integer"
+      })
+      .int({
+        message: "availableSeats must be an integer"
+      })
+      .nonnegative({
+        message: "availableSeats must be non-negative"
+      })
+      .optional(),
+
+    activePosts: z
+      .boolean({
+        invalid_type_error: "activePosts must be a boolean"
+      })
+      .optional(),
+
+    startAtPost: z
+      .number({
+        invalid_type_error: "startAtPost must be an integer"
+      })
+      .int({
+        message: "startAtPost must be an integer"
+      })
+      .positive({
+        message: "startAtPost must be positive"
+      })
+      .optional(),
+
+    endAtPost: z
+      .number({
+        invalid_type_error: "endAtPost must be an integer"
+      })
+      .int({
+        message: "endAtPost must be an integer"
+      })
+      .positive({
+        message: "endAtPost must be positive"
+      })
+      .optional(),
+
+    orderBy: z
+      .number({
+        invalid_type_error: "orderBy must be an integer"
+      })
+      .int({
+        message: "orderBy must be an integer"
+      })
+      .gte(-3, {
+        message: "orderBy must be an integer in the range [-3,3]"
+      })
+      .lte(3, {
+        message: "orderBy must be an integer in the range [-3,3]"
+      })
+      .optional()
+
+  })
+    .refine(data => (!(data.startTime && data.endTime) || (new Date(data.startTime) < new Date(data.endTime))),
+      "startTime must occur before endTime",
+    )
+})
+
+export const searchPostValidator = validate(dataSchema)
+
+let orderingBy: object = {
+  1: "post.createdAt",
+  2: "post.timeRangeStart",
+  3: "post.seats",
+}
+
+let orderingAlong: object = {
+  0: "ASC",
+  1: "DESC",
+  2: "DESC",
+}
+
+export const searchPosts = async (req: Request, res: Response) => {
+
+  let fromPlace: Place | null = req.body.fromPlace;
+  let toPlace: Place | null = req.body.toPlace;
+  let startTime: Date | null = req.body.startTime; // Renders trips whose timeRange is within or after startTime
+  let endTime: Date | null = req.body.endTime; // Renders trips whose timeRange is within or before endTime
+  // Use 1 or more here, to show only those posts which have available seats. leaving empty renders all posts without checking seats
+  let availableSeats: number = req.body.availableSeats || 0;
```

Comment:
Shouldn't this be defaulted to 1 instead? Won't that make more sense

Reply:
it shows all posts in case u need to search for filled rides too. The main aim was to make it as flexible as possible.

Reply:
how is this more flexible than keeping default 1?

Reply:
Cool its just better to keep this condition if it's availed, otherwise lets just remove it.